hudi-cli/src/main/java/org/apache/hudi/cli/commands/SparkMain.java

```diff
@@ -263,13 +265,26 @@ private static int compact(JavaSparkContext jsc, String basePath, String tableNa
   }
 
   private static int deduplicatePartitionPath(JavaSparkContext jsc, String duplicatedPartitionPath,
-      String repairedOutputPath, String basePath, String dryRun) {
+      String repairedOutputPath, String basePath, boolean dryRun, String dedupeType) {
     DedupeSparkJob job = new DedupeSparkJob(basePath, duplicatedPartitionPath, repairedOutputPath, new SQLContext(jsc),
-        FSUtils.getFs(basePath, jsc.hadoopConfiguration()));
-    job.fixDuplicates(Boolean.parseBoolean(dryRun));
+        FSUtils.getFs(basePath, jsc.hadoopConfiguration()), getDedupeType(dedupeType));
+    job.fixDuplicates(dryRun);
     return 0;
   }
 
+  private static Enumeration.Value getDedupeType(String type) {
+    switch (type) {
+      case "insertType":
+        return DeDupeType.insertType();
+      case "updateType":
+        return DeDupeType.updateType();
+      case "upsertType":
+        return DeDupeType.upsertType();
+      default:
+        throw new IllegalArgumentException("Please provide valid dedupe type!");
+    }
+  }
+
```

Comment:
Can use `DeDupeType.withName("insertType")` instead?

Reply:
But what difference does it create? 

`DeDupeType.insertType()` and `DeDupeType.withName("insertType")` - both return the same Value.

Reply:
@pratyakshsharma : I mean that we can use `DeDupeType.withName("insertType")` to convert `String` to `Enum`.  `getDedupeType` Function may not need here.

Reply:
Taken care. :)